File: tests/csharp_sample_project/AppConfig.cs
// tests/sample_project2/AppConfig.cs

namespace SampleProject2;

/// <summary>
/// Stores configuration settings used by other parts of the application.
/// </summary>
public static class AppConfig
{
    // --- Constants for Configuration ---

    /// <summary>
    /// Simulates a path to a data file (used by DataHandler).
    /// </summary>
    public const string DataFilePath = "data/items.json";

    /// <summary>
    /// A processing parameter (used by ItemProcessor).
    /// </summary>
    public const int ProcessingThreshold = 100;

    /// <summary>
    /// Example setting for logging level.
    /// </summary>
    public const string LogLevel = "INFO";

    /// <summary>
    /// Returns the configured path for the data file.
    /// </summary>
    public static string GetDataPath()
    {
        Console.WriteLine($"Config: Providing data file path: {DataFilePath}");
        return DataFilePath;
    }

    /// <summary>
    /// Returns the configured processing threshold.
    /// </summary>
    public static int GetThreshold()
    {
        Console.WriteLine($"Config: Providing processing threshold: {ProcessingThreshold}");
        return ProcessingThreshold;
    }
}

File: tests/swift_sample_project/SampleProject2/AppConfig.swift
// Sources/SampleProject2/AppConfig.swift

import Foundation

/// Stores configuration settings used by other parts of the application.
public enum AppConfig {
    
    /// Simulates a path to a data file (used by DataHandler).
    public static let dataFilePath = "data/items.json"
    
    /// A processing parameter (used by ItemProcessor).
    public static let processingThreshold = 100
    
    /// Example setting for logging level.
    public static let logLevel = "INFO"
    
    /// Returns the configured path for the data file.
    public static func getDataPath() -> String {
        print("Config: Providing data file path: \(dataFilePath)")
        return dataFilePath
    }
    
    /// Returns the configured processing threshold.
    public static func getThreshold() -> Int {
        print("Config: Providing processing threshold: \(processingThreshold)")
        return processingThreshold
    }
}